client/features/api.tsx

```diff
@@ -4,40 +4,56 @@ import { RootState } from '../app/store';
 
 // Define a service using a base URL and expected endpoints
 export const api = createApi({
-    reducerPath: 'api',
-    baseQuery: fetchBaseQuery({
-        baseUrl: 'http://localhost:3000/api', 
-        prepareHeaders: (headers: Headers, { getState }) => {
-          const token = (getState() as RootState).auth.token;
-          if (token) {
-            headers.set("Authorization", `Bearer ${token}`);
-          }
-          return headers;
-        },
-      }),
-      tagTypes: ['CurrentUser', 'Zoop', 'Fave'],
-      endpoints: (builder) => ({
-        register: builder.mutation({
-          query: ({ email, username, password }) => ({
-            url: "users/register",
-            method: "POST",
-            body: { email, username, password },
-          }),
-          invalidatesTags: ["CurrentUser"],
+  reducerPath: 'api',
+  baseQuery: fetchBaseQuery({
+      baseUrl: 'http://localhost:3000/api', 
+      prepareHeaders: (headers: Headers, { getState }) => {
+        const token = (getState() as RootState).auth.token;
+        if (token) {
+          headers.set("Authorization", `Bearer ${token}`);
+        }
+        return headers;
+      },
+    }),
+    tagTypes: ['CurrentUser', 'Zoop', 'Fave', 'User'],
+    endpoints: (builder) => ({
+      register: builder.mutation({
+        query: ({ email, username, password }) => ({
+          url: "users/register",
+          method: "POST",
+          body: { email, username, password },
         }),
-        login: builder.mutation({
-          query: ({ email, password }) => ({
-            url: "users/login",
-            method: "POST",
-            body: { email, password },
-          }),
-          invalidatesTags: ["CurrentUser"],
+        invalidatesTags: ["CurrentUser"],
+      }),
+      login: builder.mutation({
+        query: ({ email, password }) => ({
+          url: "users/login",
+          method: "POST",
+          body: { email, password },
         }),
-        getAllZoops: builder.query<{zoops: Zoop[]}, void>({
-          query: () => `/zoops`,
-          providesTags: ['Zoop']
+        invalidatesTags: ["CurrentUser"],
+      }),
+      getAllZoops: builder.query<{zoops: Zoop[]}, void>({
+        query: () => `/zoops`,
+        providesTags: ['Zoop']
+      }),
+      postZoop: builder.mutation({
```

Comment:
On the front-end, this is typically referred to as a create action (POST is just the name of the HTTP method). I would recommend renaming either to createZoop or, to reflect the UX better, probably sendZoop

```suggestion
      sendZoop: builder.mutation({
```